### ISSUE TITLE
When the disk drive is full, NVDA can't write it's log, which results in endless errors

### ISSUE THREAD

**Issue by zahyur** [nvaccess/nvda#6330]
I forgot about this, but I was reminded by a recent discussion on another issue.
A while ago it happened to me to have a 100% full Windows system drive.
That resulted in constant NVDA errors. I used a snapshot back then, so NVDA became unusable. I'm not sure if that would be the case with a stable version, where we don't have the error sound.
I don't keep a log, and I'm not sure there was one, but I'm almost sure it would have been a result of NVDA been unable to write it's log - I can't thing of another place, which could have such effect (not that there could not be one).
I can't look at NVDA code right now, but if it doesn't guard against this, I suggest adding such a check.

**Comment by zahyur** [comment]
Obviously, I probably have had the log level set to a more verbose option (I usually set it to debug).
If you have it set to a less aggressive level, like info, you probably wouldn't experience this, i.e. you'll get a few errors and that would be it.

**Comment by zahyur** [comment]
Or probably not - it would be an exception, so it'll try to log it every time, regardless the logging level, which will generate another exception and so on.

**Comment by feerrenrut** [comment]
P3 Given this seems like a fairly rare issue to run into, however it could be handled more gracefully. Assuming the implementation time to fix this is fairly low based on catching an "unable to write" error and disabling logging.

**Comment by Adriani90** [comment]
Disabling logging is now possible in NVDA 2019.1 Beta. @zahyur, is this issue now still significant even with disabled logging?

**Comment by Brian1Gaff** [comment]
Hmm, any drive that full, though is not going to work very well with windoze 
is it? :-)

However i wonder if disabling the log might fix my issue when nvda won't 
restart after an update, as it is clear that it is trying to write to the 
log that creates the problem if the log file is still locked.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by lukaszgo1** [comment]
I am not @zahyur, but in my view it still is significant. Imagine the following I've actually experienced it a few times when I had small SSD in my laptop:
1. User starts downloading something big some torrent containing a few iso's for various distros or something similar.
2. For some reason or another he has to restart NVDA during download without realizing, that it makes his c drive almost full.
3. Even with the logging disabled the first lines of the log needs to be written, because until configuration is read there is no possibility to realize that logging was disabled.
4. Writing anything fails, because there is no space.
In case of a release version there is simply a Windows error sound and NVDa fails to start
With snapshots the error sound plays continually until space is somehow found.
I've tried fixing this half  year ago, but sadly failed. I  will probably look into this again when time permits.
